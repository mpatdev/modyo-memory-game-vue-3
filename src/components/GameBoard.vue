<script setup>
import GameCard from '@/components/GameCard.vue'
import useModyoApi from '@/composables/useModyoApi'
import useCardGame from '@/composables/useCardGame'

const { images, refresh } = useModyoApi()
const { cards, 
  shuffleCards, 
  flipCard, 
  isFlipped, 
  currentFlipped, 
  isFound 
} = useCardGame(images)

async function startGame() {
  
  await refresh()

  shuffleCards()

  console.log('Iniciando Juego')
}
</script>

<template>
  <div>
    <div>
      <button @click="startGame">
        Iniciar Juego
      </button>
      {{ currentFlipped }}
    </div>
    <div class="grid grid-cols-5 grid-rows-4 gap-4 h-[100dvh]">
      <GameCard
        v-for="card in cards"
        :key="card.id"
        :item="card"
        :is-flipped="isFlipped(card)"
        :is-found="isFound(card)"
        @click="() => flipCard(card)"
      />
    </div>
  </div>
</template>