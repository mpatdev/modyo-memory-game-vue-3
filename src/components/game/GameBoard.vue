<script setup>
import GameCard from '@/components/game/GameCard.vue'
import useModyoApi from '@/composables/useModyoApi'
import useCardGame from '@/composables/useCardGame'

const { images, refresh } = useModyoApi()
const { cards, 
  shuffleCards, 
  flipCard, 
  isFlipped,
  isFound
} = useCardGame(images)

async function startGame() {
  
  await refresh()

  shuffleCards()

  console.log('Iniciando Juego')
}
</script>

<template>
  <div>
    <div>
      <button @click="startGame">
        Iniciar Juego
      </button>
    </div>
    <div class="grid grid-cols-5 grid-rows-4 gap-4 h-[100dvh]">
      <GameCard
        v-for="card in cards"
        :key="card.id"
        :item="card"
        :is-flipped="isFlipped(card)"
        :is-found="isFound(card)"
        @click="() => flipCard(card)"
      />
    </div>
  </div>
</template>