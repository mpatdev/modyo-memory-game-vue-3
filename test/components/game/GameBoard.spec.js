import GameBoard from '@/components/game/GameBoard.vue'

import snapshotTest from '@test/snapshotTest'

snapshotTest(GameBoard, 'GameBoard')