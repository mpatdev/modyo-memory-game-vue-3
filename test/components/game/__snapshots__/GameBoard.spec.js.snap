// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameBoard > Check Snapshot 1`] = `
"<div class=\\"game-board\\">
  <div class=\\"flex w-full h-full justify-center items-center flex-col gap-4\\">
    <!--v-if--><button role=\\"button\\" class=\\"game-button\\"> Start Game </button>
  </div>
  <!--v-if-->
</div>"
`;
